apps/site/lib/site_web/controllers/schedule/line_api.ex

```diff
@@ -151,15 +151,14 @@ defmodule SiteWeb.ScheduleController.LineApi do
     Enum.any?(alerts, &Alerts.Alert.is_diversion(&1))
   end
 
-  # for each list of disrupted stops,
-  # make adjustment to the diagram
-  # based on the nature of the disruption.
-  # e.g. for detour we modify the stop LEADING to it
-  # but for shuttle we DON'T modify the FINAL stop
-  # note: there might be more logic than that.
+  # for each list of disrupted stops, make adjustment to the diagram based on
+  # the nature of the disruption. e.g. for detour or stop/station closure we
+  # modify the stop preceding it but for shuttle we don't style the final stop
   defp shift_indices_by_disruption_effect(indices, effects) do
+    effect_avoids_stop? = effects -- effects -- [:stop_closure, :station_closure, :detour]
```

Comment:
`effect_avoids_stop?` isn't actually a boolean as the result of this, which is causing the `and` problem below. Would something like this be equivalent but result in a boolean value?

```suggestion
    effect_avoids_stop? = Enum.any?(effects, &Enum.member?([:stop_closure, :station_closure, :detour], &1))
```

Alternately, you could also do something like `effects -- effects -- [:stop_closure, :station_closure, :detour] != []`.

Reply:
(See later comment where I see you updated this further…)

Reply:
I like your proposed change better so will update it again :)